src/EventStore.ClientAPI/Internal/ClusterDnsEndPointDiscoverer.cs

```diff
@@ -109,17 +112,31 @@ internal class ClusterDnsEndPointDiscoverer : IEndPointDiscoverer {
 			return null;
 		}
 
-		private GossipSeed[] GetGossipCandidatesFromDns() {
+		private IGossipSeed[] GetGossipCandidatesFromConfig() {
 			//_log.Debug("ClusterDnsEndPointDiscoverer: GetGossipCandidatesFromDns");
-			GossipSeed[] endpoints;
-			if (_gossipSeeds != null && _gossipSeeds.Length > 0) {
-				endpoints = _gossipSeeds;
+
+			// Safe in this case.
+			IGossipSeed[] endpoints = _gossipSeeds;
+
+			if ((endpoints?.Length ?? 0) == 0) {
+				if (_compatibilityMode.IsAutoCompatibilityModeEnabled()) {
+					endpoints = new[] {
+						new ClusterDnsSeed(_clusterDns, _managerExternalHttpPort, seedOverTls: true, v20HostHeader: true), // Then HTTPS gossip first (v20 defaults)
+						new ClusterDnsSeed(_clusterDns, _managerExternalHttpPort, seedOverTls: false, v20HostHeader: false), // Try HTTP gossip (v5 defaults)
```

Comment:
Wouldn't you expect the v5 client (this branch) to try the `v5 default` `ClusterDnsSeed` first?  Doesn't make too much difference I guess given it'll quickly move onto it but might be clearer.

Reply:
We decided to lean v20 by default when enabling auto compatibility mode.